scripts/process_data.py

```diff
@@ -0,0 +1,234 @@
+known_status_file = os.path.join('data', 'status_matrix.tsv')
+
+
+# ## Load Data
+
+# In[4]:
+
+rnaseq_df = pd.read_table(rna_file, index_col=0)
+mutation_df = pd.read_table(mut_file)
+copy_df = pd.read_table(copy_file, index_col=0)
+clinical_df = pd.read_table(clinical_file, index_col=0, low_memory=False)
+
+
+# ## Begin processing different data types
+# 
+# ### RNAseq
+
+# In[5]:
+
+# Process RNAseq file
+rnaseq_df.index = rnaseq_df.index.map(lambda x: x.split('|')[0])
+rnaseq_df.columns = rnaseq_df.columns.str.slice(start=0, stop=15)
+rnaseq_df = rnaseq_df.drop('?').fillna(0).sort_index(axis=1)
+
+# Gene is listed twice in RNAseq data, drop both occurrences
+rnaseq_df.drop('SLC35E2', axis=0, inplace=True)
+rnaseq_df = rnaseq_df.T
+
+# Determine most variably expressed genes and subset
+num_mad_genes = 5000
+mad_genes = rnaseq_df.mad(axis=0).sort_values(ascending=False)
+top_mad_genes = mad_genes.iloc[0:num_mad_genes, ].index
+rnaseq_subset_df = rnaseq_df.loc[:, top_mad_genes]
+
+
+# In[6]:
+
+# Scale RNAseq data using z-scores
+rnaseq_scaled_df = preprocessing.StandardScaler().fit_transform(rnaseq_subset_df)
+rnaseq_scaled_df = pd.DataFrame(rnaseq_scaled_df,
+                                columns=rnaseq_subset_df.columns,
+                                index=rnaseq_subset_df.index)
+rnaseq_scaled_df.to_csv(rna_out_file, sep='\t')
+
+# Scale RNAseq data using zero-one normalization
+rnaseq_scaled_zeroone_df = preprocessing.MinMaxScaler().fit_transform(rnaseq_subset_df)
+rnaseq_scaled_zeroone_df = pd.DataFrame(rnaseq_scaled_zeroone_df,
+                                        columns=rnaseq_subset_df.columns,
+                                        index=rnaseq_subset_df.index)
+rnaseq_scaled_zeroone_df.to_csv(rna_out_zeroone_file, sep='\t')
+
+
+# ### Mutation
+
+# In[7]:
+
+# Filter mutation types and generate binary matrix
+mutations = {
+    'Frame_Shift_Del',
+    'Frame_Shift_Ins',
+    'In_Frame_Del',
+    'In_Frame_Ins',
+    'Missense_Mutation',
+    'Nonsense_Mutation',
+    'Nonstop_Mutation',
+    'RNA',
+    'Splice_Site',
+    'Translation_Start_Site',
+}
+
+
+# In[8]:
+
+# Process mutation in long format to dataframe format
+mut_pivot = (mutation_df.query("effect in @mutations")
+                        .groupby(['#sample', 'chr',
+                                  'gene'])
+                        .apply(len).reset_index()
+                        .rename(columns={0: 'mutation'}))
+
+mut_pivot = (mut_pivot.pivot_table(index='#sample',
+                                   columns='gene', values='mutation',
+                                   fill_value=0)
+                      .astype(bool).astype(int))
+
+mut_pivot.to_csv(mut_out_file, sep='\t')
+
+
+# ### Copy Number
+
+# In[9]:
+
+copy_df = copy_df.fillna(0)
+copy_df = copy_df.astype(int)
+copy_df = copy_df.T
+copy_df.columns.name = 'gene'
+copy_df.index.name = 'Sample'
+
+
+# In[10]:
+
+copy_loss_df = copy_df.replace(to_replace=[1, 2, -1], value=0)
```

Comment:
Can you add a comment line explaining why you are replacing these values with 0 or at least explain this these lines to me?